src/System.Diagnostics.Debug.SymbolReader/src/System/Diagnostics/Debug/SymbolReader/SymbolReader.cs

```diff
@@ -85,30 +84,20 @@ public static bool GetLocalVariableName(string assemblyFileName, int methodToken
 
             MethodDebugInformationHandle methodDebugHandle = ((MethodDefinitionHandle)handle).ToDebugInformationHandle();
             LocalScopeHandleCollection localScopes = pdbReader.GetLocalScopes(methodDebugHandle);
-            LocalScope? bestScope = null;
             foreach (LocalScopeHandle scopeHandle in localScopes)
             {
                 LocalScope scope = pdbReader.GetLocalScope(scopeHandle);
-                if (!bestScope.HasValue)
-                    bestScope = scope;
-                else if (scope.StartOffset > ilOffset)
-                    break;
-                else if (scope.StartOffset >= bestScope.Value.StartOffset)
-                    bestScope = scope;
-            }
-            
-            if (!bestScope.HasValue)
-                return false;
-            LocalVariableHandleCollection localVars = bestScope.Value.GetLocalVariables();
-            foreach (LocalVariableHandle varHandle in localVars)
-            {
-                LocalVariable localVar = pdbReader.GetLocalVariable(varHandle);
-                if (localVar.Index == localIndex)
+                LocalVariableHandleCollection localVars = scope.GetLocalVariables();
+                foreach (LocalVariableHandle varHandle in localVars)
                 {
-                    if (localVar.Attributes == LocalVariableAttributes.DebuggerHidden)
-                        return false;
-                    localVarName = Marshal.StringToBSTR(pdbReader.GetString(localVar.Name));
-                    return true;
+                    LocalVariable localVar = pdbReader.GetLocalVariable(varHandle);
+                    if (localVar.Index == localIndex)
+                    {
+                        if (localVar.Attributes == LocalVariableAttributes.DebuggerHidden)
+                            return false;
+                        localVarName = Marshal.StringToBSTR(pdbReader.GetString(localVar.Name));
```

Comment:
It would be nice to separate the actual logic from marshalling details. Perhaps factor out a helper method that doesn't do any marshalling and then marshal the outputs of that method as needed.